test/e2e/upg_e2e.go

```diff
@@ -604,6 +604,59 @@ var _ = ginkgo.Describe("CLI debug commands", func() {
 
 // TODO: validate both binapi and CLI against each other
 var _ = ginkgo.Describe("UPG Binary API", func() {
+	ginkgo.Context("for upf tdf ul table", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+
+		addMapping := func(isIPv6 bool, src uint32, dst uint32) {
+			req := &upf.UpfTdfUlTableAdd{
+				IsAdd:            true,
+				IsIPv6:           isIPv6,
+				TableID:          src,
+				SrcLookupTableID: dst,
+			}
+			reply := &upf.UpfTdfUlTableAddReply{}
+			gomega.Expect(
+				f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply),
+			).To(gomega.Succeed(), "upf_tdf_ul_table_add")
+		}
+
+		getMappings := func(isIPv6 bool) *upf.UpfTdfUlTableReply {
+			req := &upf.UpfTdfUlTable{
+				IsIPv6: isIPv6,
+			}
+			reply := &upf.UpfTdfUlTableReply{}
+			gomega.Expect(
+				f.VPP.ApiChannel.SendRequest(req).ReceiveReply(reply),
+			).To(gomega.Succeed(), "upf_tdf_ul_table")
+			return reply
+		}
+
+		ginkgo.It("sets and retrieves table mappings", func() {
+			// there should be no ipv6 mappings initially
+			mappings := getMappings(true)
+			gomega.Expect(mappings.MappingsLen).To(gomega.Equal(uint8(0)), "upf_tdf_ul_table")
+			gomega.Expect(mappings.Mappings).To(gomega.Equal([]uint32{}), "upf_tdf_ul_table")
```

Comment:
After fixing the comments above, this could be written as
```suggestion
			gomega.Expect(getMappings()).To(gomega.Equal([]uint32{}), "upf_tdf_ul_table")
```

Reply:
The same changes can be done in the checks below